src/base/utils/net.cpp

```diff
@@ -38,5 +40,42 @@ namespace Utils
         {
             return !QHostAddress(ip).isNull();
         }
+
+        bool isValidSubnet(const QPair<QHostAddress, int> &subnet)
+        {
+            const QPair<QHostAddress, int> invalid = qMakePair(QHostAddress(), -1);
+            return subnet != invalid;
+        }
+
+        bool isLoopbackAddress(const QHostAddress &addr)
+        {
+            return addr == QHostAddress::LocalHost
+                    || addr == QHostAddress::LocalHostIPv6
+                    || addr == QHostAddress(QLatin1String("::ffff:127.0.0.1"));
+        }
+
+        bool isIPInSubnets(const QHostAddress &addr, const QList<QPair<QHostAddress, int>> &subnets)
```

Comment:
IMO, I would have it another way. I would have isIPInSubnet() function here since it more common, and use it in loop where it needed.

Reply:
`QHostAddress::isInSubnet` already handles that functionality. This is a bespoke method that abstracts away the ipv4/ipv6 conversion and whether it was successful, while only performing the conversion once for the entire list. If anything this should probably be moved out of utils.

Reply:
>only performing the conversion once for the entire list.

Ok. Leyt it be as-is.